src/app/plugins/help.plugin.ts

```diff
@@ -1,35 +1,97 @@
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { IContainer, IMessage, ChannelType, IPluginHelp } from '../../common/types';
 import Constants from '../../common/constants';
 import { RichEmbed } from 'discord.js';
 
 export class HelpPlugin extends Plugin {
   public name: string = 'Help Plugin';
   public description: string = 'Displays supported commands and usage statements.';
-  public usage: string = 'help';
+  public usage: string = 'help [Plugin Command]';
+  public pluginAlias = [];
   public permission: ChannelType = ChannelType.Bot;
-  private _embed: RichEmbed = new RichEmbed();
+  private _embed: RichEmbed[] = [];
+  private _pluginEmbed: IPluginHelp = {};
+
   constructor(public container: IContainer) {
     super();
+
+    this._embed[0] = new RichEmbed();
+    this._embed[1] = new RichEmbed();
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    if (this._embed.fields!.length === 0) {
-      this._generateEmbed();
+    const commands = this.container.pluginService.aliases;
+    const input: string = this._parseCommand(args || []);
+
+    if (!!commands[input]) {
+      const pluginName = commands[input];
+      const plugin = this.container.pluginService.plugins[pluginName];
+
+      if (plugin.permission === ChannelType.Admin || plugin.permission === ChannelType.Staff) {
+        if (this._embed[1].fields!.length === 0) {
+          this._generateEmbed(false, 1);
+        }
+
+        message.reply(this._embed[1]);
+      } else {
+        if (this._pluginEmbed[pluginName] === undefined) {
+          this._pluginEmbed[pluginName] = new RichEmbed();
+          this._generatePluginEmbed(pluginName);
+        }
+        message.reply(this._pluginEmbed[pluginName]);
+      }
+    } else if (input === `all`) {
+      if (this._embed[0].fields!.length === 0) {
+        this._generateEmbed(true, 0);
+      }
+
+      message.reply(this._embed[0]);
+    } else {
+      if (this._embed[1].fields!.length === 0) {
+        this._generateEmbed(false, 1);
+      }
+
+      message.reply(this._embed[1]);
     }
-    message.reply(this._embed);
   }
 
-  private _generateEmbed() {
+  private _generateEmbed(all: boolean, targ: number) {
     const plugins = Object.keys(this.container.pluginService.plugins);
 
-    this._embed.setColor('#0099ff').setTitle('**__These are the commands I support__**');
+    this._embed[targ].setColor('#0099ff').setTitle('**__These are the commands I support__**');
+
+    for (const targName of plugins) {
+      const plugin = this.container.pluginService.get(targName);
+      const aliases = plugin.pluginAlias || [];
 
-    for (let i = 0; i < plugins.length; i++) {
-      const plugin = this.container.pluginService.get(plugins[i]);
+      const altCalls =
+        all && aliases.length != 0 ? `aliases: ${aliases.join(', ')} , \n` : `aliases: None \n`;
```

Comment:
common prefix of "aliases: " should be outside the logic. Also, `aliases.length && aliases.join() || "None"`